daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_functions/BitFunctionsNot.tdml

```diff
@@ -0,0 +1,305 @@
+      <tdml:infoset>
+        <tdml:dfdlInfoset>
+          <ex:testShortNot>
+            <ex:ShortNot1>
+              <num>1</num>
+              <value>-2</value>
+            </ex:ShortNot1>
+            <ex:ShortNot2>
+              <num>0</num>
+              <value>-1</value>
+            </ex:ShortNot2>
+          </ex:testShortNot>
+        </tdml:dfdlInfoset>
+      </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="testByteNot" root="testByteNot" model="BitFunctions">
+    <tdml:document>1|0</tdml:document>
+      <tdml:infoset>
+        <tdml:dfdlInfoset>
+          <ex:testByteNot>
+            <ex:ByteNot1>
+              <num>1</num>
+              <value>-2</value>
+            </ex:ByteNot1>
+            <ex:ByteNot2>
+              <num>0</num>
+              <value>-1</value>
+            </ex:ByteNot2>
+          </ex:testByteNot>
+        </tdml:dfdlInfoset>
+      </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="testUnsignedIntNot" root="testUnsignedIntNot" model="BitFunctions">
+    <tdml:document>2|1</tdml:document>
+      <tdml:infoset>
+        <tdml:dfdlInfoset>
+          <ex:testUnsignedIntNot>
+            <ex:UnsignedIntNot1>
+              <num>2</num>
+              <value>4294967293</value>
+            </ex:UnsignedIntNot1>
+            <ex:UnsignedIntNot2>
+              <num>1</num>
+              <value>4294967294</value>
+            </ex:UnsignedIntNot2>
+          </ex:testUnsignedIntNot>
+        </tdml:dfdlInfoset>
+      </tdml:infoset>
+  </tdml:parserTestCase>
+
+
+  <tdml:parserTestCase name="testUnsignedLongNot" root="testUnsignedLongNot" model="BitFunctions">
+    <tdml:document>2|1</tdml:document>
+      <tdml:infoset>
+        <tdml:dfdlInfoset>
+          <ex:testUnsignedLongNot>
+            <ex:UnsignedLongNot1>
+              <num>2</num>
+              <value>-3</value>
+            </ex:UnsignedLongNot1>
+            <ex:UnsignedLongNot2>
+              <num>1</num>
+              <value>-2</value>
+            </ex:UnsignedLongNot2>
+          </ex:testUnsignedLongNot>
+        </tdml:dfdlInfoset>
+      </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="testUnsignedShortNot" root="testUnsignedShortNot" model="BitFunctions">
+    <tdml:document>2|1</tdml:document>
+      <tdml:infoset>
+        <tdml:dfdlInfoset>
+          <ex:testUnsignedShortNot>
+            <ex:UnsignedShortNot1>
+              <num>1</num>
+              <value>4294967293</value>
+            </ex:UnsignedShortNot1>
+            <ex:UnsignedShortNot2>
+              <num>1</num>
+              <value>4294967294</value>
+            </ex:UnsignedShortNot2>
+          </ex:testUnsignedShortNot>
+        </tdml:dfdlInfoset>
+      </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="testUnsignedByteNot" root="testUnsignedByteNot" model="BitFunctions">
+    <tdml:document>-1|6</tdml:document>
```

Comment:
-1 isn't a valid unsignedByte so this will fail to parse. The test also expects the bitwise-not of 6 is -7, which isn't correct for unsigned bytes.